specs-go/config.go

```diff
@@ -0,0 +1,49 @@
+package specs
+
+// Spec is the base configuration for CDI
+type Spec struct {
+	Version          string   `json:"cdiVersion"`
+	Kind             string   `json:"kind"`
+	KindShort        []string `json:"kindShort,omitempty"`
+	ContainerRuntime []string `json:"containerRuntime,omitempty"`
+
+	Devices  []Devices `json:"devices"`
+	OCIEdits OCIEdits  `json:"ociEdits,omitempty"`
+}
+
+// Devices is a "Device" a container runtime can add to a container
+type Devices struct {
+	Name      string   `json:"name"`
+	NameShort []string `json:"nameShort"`
+	OCIEdits  OCIEdits `json:"ociEdits"`
+}
+
+// OCIEdits are edits a container runtime must make to the OCI spec to expose the device.
+type OCIEdits struct {
+	DeviceNodes []DeviceNode `json:"deviceNodes,omitempty"`
+	Hooks       []Hook       `json:"hooks,omitempty"`
+	Mounts      []Mount      `json:"mounts,omitempty"`
+}
+
+// DeviceNode represents a device node that needs to be added to the OCI spec.
+type DeviceNode struct {
+	HostPath      string `json:"hostPath"`
+	ContainerPath string `json:"containerPath"`
+	Permissions   string `json:"permissions,omitempty"`
+}
```

Comment:
Would it make sense to use types from OCI runtime spec as much as possible?
For example we can use [LInuxDevice structure](https://github.com/opencontainers/runtime-spec/blob/master/specs-go/config.go#L377:L393) here.

The overall idea is to simplify updating OCI spec as far as I understood. Reusing spec structures would make it much easier I believe.

Reply:
Sorry for the very late reply, reusing the `LinuxDevice` structure has the unfortunate consequence that it makes it Linux specific. With `HostPath`/`ContainerPath` we keep this more or less OS agnostic.